apps/documents/models.py

```diff
@@ -24,3 +26,6 @@ def size(self) -> float:
 
     def file_names(self) -> list[str]:
         return list(self.files.values_list("name", flat=True))
+
+    def get_references(self) -> list[Node]:
+        return Node.objects.filter(type=LLMResponseWithPrompt.__name__, params__collection_id=str(self.id)).all()
```

Comment:
Probably worth making a manager method for this query since we do it in a few places now.

Reply:
Also:
```suggestion
        return Node.objects.select_related("pipeline").filter(type=LLMResponseWithPrompt.__name__, params__collection_id=str(self.id)).all()
```